packages/client/modules/userDashboard/components/OrgBilling/UpdatePayment.tsx

```diff
@@ -33,14 +50,119 @@ const CARD_ELEMENT_OPTIONS = {
+  handleClose: () => void
+  orgId: string
 }
 
 const UpdatePayment = (props: Props) => {
-  const {handleCancel} = props
+  const {handleClose, orgId} = props
+  const atmosphere = useAtmosphere()
+  const {onError, onCompleted} = useMutationProps()
+  const [isLoading, setIsLoading] = useState(false)
+  const stripe = useStripe()
+  const elements = useElements()
+  const [errorMsg, setErrorMsg] = useState<null | string>()
+  const [hasStarted, setHasStarted] = useState(false)
+  const [cardNumberError, setCardNumberError] = useState<null | string>()
+  const [expiryDateError, setExpiryDateError] = useState<null | string>()
+  const [cvcError, setCvcError] = useState<null | string>()
+  const [cardNumberComplete, setCardNumberComplete] = useState(false)
+  const [expiryDateComplete, setExpiryDateComplete] = useState(false)
+  const [cvcComplete, setCvcComplete] = useState(false)
+  const hasValidCCDetails =
+    cardNumberComplete &&
+    expiryDateComplete &&
+    cvcComplete &&
+    !cardNumberError &&
+    !expiryDateError &&
+    !cvcError
+  const isUpdateDisabled = isLoading || !stripe || !elements || !hasValidCCDetails
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+    if (!stripe || !elements) return
+    setIsLoading(true)
+    if (errorMsg) {
+      setIsLoading(false)
+      setErrorMsg(null)
+      return
+    }
+    const cardElement = elements.getElement(CardNumberElement)
+    if (!cardElement) {
+      setIsLoading(false)
+      const newErrorMsg = 'Something went wrong. Please try again.'
+      setErrorMsg(newErrorMsg)
+      return
+    }
+    const {paymentMethod, error} = await stripe.createPaymentMethod({
+      type: 'card',
+      card: cardElement
+    })
+    if (error) {
+      setErrorMsg(error.message)
+      setIsLoading(false)
+      return
+    }
+
+    const handleCompletedUpdate = async (res: UpdateCreditCardMutation$data) => {
+      const {updateCreditCard} = res
+      const {stripeSubscriptionClientSecret, error} = updateCreditCard
+      if (error || !stripeSubscriptionClientSecret) {
+        const newErrMsg =
+          error?.message ?? 'Something went wrong. Please try again or contact support.'
+        setIsLoading(false)
+        setErrorMsg(newErrMsg)
+        return
+      }
+      const {error: confirmationError} = await stripe.confirmCardPayment(
+        stripeSubscriptionClientSecret
+      )
+      setIsLoading(false)
+      if (confirmationError) {
+        setErrorMsg(confirmationError.message)
+        return
+      }
+      onCompleted()
+      handleClose()
+    }
+
+    UpdateCreditCardMutation(
+      atmosphere,
+      {orgId, paymentMethodId: paymentMethod.id},
+      {onError, onCompleted: handleCompletedUpdate}
+    )
+  }
+
+  const handleChange =
+    (type: 'CardNumber' | 'ExpiryDate' | 'CVC') => (event: StripeElementChangeEvent) => {
+      if (errorMsg) setErrorMsg(null)
+      if (!hasStarted && !event.empty) {
```

Comment:
-1 this appears to be the only place where `hasStarted` is read, please remove or add a comment why it's needed